src/charm.py

```diff
@@ -139,6 +140,18 @@ def __init__(self, *args):
         self.framework.observe(self.on[PEER_RELATION_NAME].relation_created, self._configure)
         self.framework.observe(self.on[PEER_RELATION_NAME].relation_changed, self._configure)
 
+    @contextmanager
+    def temp_maintenance_status(self: CharmBase, message: str):
```

Comment:
Just a nit, but we haven't typed `self` anywhere else than here

Reply:
yeah it's probably not necessary

Reply:
fixed in https://github.com/canonical/vault-operator/pull/80/commits/8a0558733da38eb41b385a7e453b1c6c1eb37be9